main/wpa_supplicant/wpa_supplicant.initd

```diff
@@ -5,7 +5,14 @@
 command=/sbin/wpa_supplicant
 : ${wpa_supplicant_conf:=/etc/wpa_supplicant/wpa_supplicant.conf}
 wpa_supplicant_if=${wpa_supplicant_if:+-i}$wpa_supplicant_if
-command_args="$wpa_supplicant_args -B -c$wpa_supplicant_conf $wpa_supplicant_if"
+
+# Run in dbus mode if no config file is found
+if [ -f "${wpa_supplicant_conf}" ]; then
+	command_args="$wpa_supplicant_args -B -c$wpa_supplicant_conf $wpa_supplicant_if"
+else
+	command_args="$wpa_supplicant_args -B -u"
+fi
+
```

Comment:
Is it possible to run wpa_supplicant in dbus mode with a config file? If so, I would recommend to have a conf.d option for dbus mode and make dbus mode default if config is missing.

Something like this:
```sh
if [ -f "${wpa_supplicant_conf}" ]; then
	command_args="${command_args} -c$wpa_supplicant_conf"
	: ${dbus_mode:=no}
else
	: ${dbus_mode:=yes}
fi

start_pre() {
	if yesno $dbus_mode; then
		command_args="${command_args} -u"
	fi
}
```

Then you can explicitly enable/disable dbus mode regardless of the existance of the wpa_supplicant config.

Otherwise you risk that you start with a no-config machine, and when you later need to set some unrelated config option for wpa_supplicant (lets say you want to disable use of some channels), you as a sideffect lose your dbus mode

Reply:
wpa_supplicant ignores the config files you supply to it if you enable dbus mode so that wouldn't be useful. `-u` overrides `-c` and `-i`

Reply:
seems like that is not the case. I think `-u` will only enable the dbus interface. Nothing more.

Reply:
Any updates on this?